cmd/srcd/cmd/parse_test.go

```diff
@@ -0,0 +1,227 @@
+	"github.com/stretchr/testify/suite"
+)
+
+type ParseTestSuite struct {
+	cmdtest.IntegrationSuite
+	testDir string
+}
+
+func TestParseTestSuite(t *testing.T) {
+	s := ParseTestSuite{}
+	suite.Run(t, &s)
+}
+
+type testCase struct {
+	path     string
+	filename string
+	lang     string
+}
+
+// Uses files from https://github.com/leachim6/hello-world
+var testCases = []testCase{
+	{
+		path:     filepath.FromSlash("../../../test/fixtures/hello.py"),
+		filename: "hello.py",
+		lang:     "python",
+	},
+	{
+		path:     filepath.FromSlash("../../../test/fixtures/hello-py3.py"),
+		filename: "hello-py3.py",
+		lang:     "python",
+	},
+	{
+		path:     filepath.FromSlash("../../../test/fixtures/hello.cpp"),
+		filename: "hello.cpp",
+		lang:     "c++",
+	},
+	{
+		path:     filepath.FromSlash("../../../test/fixtures/hello.java"),
+		filename: "hello.java",
+		lang:     "java",
+	},
+	{
+		path:     filepath.FromSlash("../../../test/fixtures/hello.js"),
+		filename: "hello.js",
+		lang:     "javascript",
+	},
+	{
+		path:     filepath.FromSlash("../../../test/fixtures/hello.bash"),
+		filename: "hello.bash",
+		lang:     "shell",
+	},
+	{
+		path:     filepath.FromSlash("../../../test/fixtures/hello.rb"),
+		filename: "hello.rb",
+		lang:     "ruby",
+	},
+	{
+		path:     filepath.FromSlash("../../../test/fixtures/hello.go"),
+		filename: "hello.go",
+		lang:     "go",
+	},
+	{
+		path:     filepath.FromSlash("../../../test/fixtures/hello.cs"),
+		filename: "hello.cs",
+		lang:     "c#",
+	},
+	{
+		path:     filepath.FromSlash("../../../test/fixtures/hello.php"),
+		filename: "hello.php",
+		lang:     "php",
+	},
+}
+
+func (s *ParseTestSuite) SetupTest() {
+}
+
+func (s *ParseTestSuite) TearDownTest() {
+	s.RunStop(context.Background())
+}
+
+func (s *ParseTestSuite) TestDriversList() {
+	require := s.Require()
+
+	out, err := s.RunCommand(context.TODO(), "parse", "drivers", "list")
+	outStr := out.String()
+
+	require.NoError(err, outStr)
+
+	/* Example output:
+
+	LANGUAGE	VERSION
+	----------	----------
+	python		v2.8.0
+	cpp		v1.1.0
+	java		v2.5.0
+	javascript	v2.6.0
+	bash		v2.4.0
+	ruby		v2.9.0
+	go		v2.5.0
+	csharp		v1.4.0
+	php		v2.7.0
+	*/
+
+	// Simple checks to see if it's the table, and contains a known driver
+	expected := regexp.MustCompile(`LANGUAGE\s+VERSION`)
+	require.Regexp(expected, outStr)
+	expected = regexp.MustCompile(`javascript\s+v\S+`)
+	require.Regexp(expected, outStr)
+}
+
+func (s *ParseTestSuite) TestLang() {
+	for _, tc := range testCases {
+		s.T().Run(tc.filename, func(t *testing.T) {
+			require := require.New(t)
+
+			// Check the language is detected
+			out, err := s.RunCommand(context.TODO(), "parse", "lang", tc.path)
+			require.NoError(err, out.String())
+			require.Equal(tc.lang+"\n", out.String())
+		})
+	}
+}
+
+// same as RunCommand, but captures only stdout instead of stdout + stderr
+func (s *ParseTestSuite) runCommandStdout(ctx context.Context, cmd string, args ...string) (*bytes.Buffer, error) {
```

Comment:
ouch :cry: 
Why not moving it parallel to the other one [IntegrationSuite::RunCommand](https://github.com/src-d/engine/blob/master/cmd/test-utils/common.go#L33), and making it public?
And even more, I'd reconsider the necessity of merging the `stderr` and `stdout`, because not doing it it could be easily ensured what is in the `stderr` or in the `stdout`, or even we could ensure that a command does not write in the `stderr`

Reply:
My 0.0000001 cents: I also noticed this, but I think it's better to improve in another PR (maybe separate stderr/out for existing tests too)

Reply:
agree on doing it in a separated issue if you prefer to handle it that way @carlosms

Reply:
yes, there are things that can be refactored after this first set of tests is done. There is also some repetition in the tmp dir initialization, and I'm sure some other things will come up. Let's take care of this in a separate PR.

Reply:
I'd love you if you also open the issue. 😘

Reply:
Done in #347